src/Envs/env_wrapper.jl

```diff
@@ -50,13 +50,7 @@ Returns the observational state of the environment. The original state can
 be accessed by `\`env._env.state\``.
 """
 function state(env::EnvWrapper)
-	try
-		return _get_obs(env._env)
-	catch y
-		if isa(y, UndefVarError) || isa(y, MethodError)
-			return env._env.state
-		end
-	end
+	return _get_obs(env._env)
```

Comment:
Any reason to get rid of the catch?

Reply:
I don't recall exactly what the error message was that caused me to remove it. Maybe Zygote can't differentiate through it?

Reply:
Ah yes we don't support try blocks in zygote yet.

But this would seem to break Gym's assumptions about what to do in case of an error here.  I'm happy to have a wrapper method that would dispatch accordingly to get rid of try block

Reply:
We could have a `_get_obs` which by default returns `env._env.state`

Reply:
Can we have that dispatched correctly and have this merged then?

Reply:
I don't understand enough about this to implement "correct dispatch", but I think you can edit the pull request.